pyrelational/datasets/uci_datasets.py

```diff
@@ -0,0 +1,194 @@
+"""
+Class to help download and do the initial processing of dataset on the UCI database
+"""
+
+import logging
+import os
+import urllib.request
+import zipfile
+from os import path
+
+import numpy as np
+import pandas as pd
+import torch
+from sklearn.model_selection import KFold, StratifiedKFold
+
+
+class UCIDatasets:
+    def __init__(self, name, data_dir="/tmp/", n_splits=10):
+        self.datasets = {
+            "concrete": "https://archive.ics.uci.edu/ml/machine-learning-databa"
+            + "ses/concrete/compressive/Concrete_Data.xls",
+            "energy": "http://archive.ics.uci.edu/ml/machine-learning-databases/00242/ENB2012_data.xlsx",
+            "power": "https://archive.ics.uci.edu/ml/machine-learning-databases/00294/CCPP.zip",
+            "wine": "https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv",
+            "yacht": "http://archive.ics.uci.edu/ml/machine-learning-databases/00243/yacht_hydrodynamics.data",
+            "glass": "https://archive.ics.uci.edu/ml/machine-learning-databases/glass/glass.data",
+            "parkinsons": "https://archive.ics.uci.edu/ml/machine-learning-databases/parkinsons/parkinsons.data",
+            "seeds": "https://archive.ics.uci.edu/ml/machine-learning-databases/00236/seeds_dataset.txt",
+            "airfoil": "https://archive.ics.uci.edu/ml/machine-learning-databases/00291/airfoil_self_noise.dat",
+        }
+        self.data_dir = data_dir
+        self.name = name
+        self.n_splits = n_splits
+
+        # flag for classification/regression dataset
+        if name in ["glass", "parkinsons", "seeds"]:
+            self.classification = True
+        else:
+            self.classification = False
+
+        self._load_dataset()
+
+    def _load_dataset(self):
+        if self.name not in self.datasets:
+            raise Exception("Not part of datasets supported in PyRelationAL at the moment")
+        if not path.exists(self.data_dir):
+            os.mkdir(self.data_dir)
+        if not path.exists(self.data_dir + "UCI"):
+            os.mkdir(self.data_dir + "UCI")
+
+        url = self.datasets[self.name]
+        file_name = url.split("/")[-1]
+        if not path.exists(self.data_dir + "UCI/" + file_name):
+            urllib.request.urlretrieve(self.datasets[self.name], self.data_dir + "UCI/" + file_name)
+        data = None
+
+        if self.name == "housing":
```

Comment:
Can there be a cleaner way than having these many if statements?

Reply:
Some of the datasets require some additional processing after downloading, so I thought this was the most straightforward way, I am not aware of a better way to handle this, cases would look the same, delegating it to another module to handle the resolution of the conditionals also would just be more code I think. I can't immediately think of a better way atm.